### ISSUE TITLE
Module not found: Error: Can't resolve 'winston-transport'

### ISSUE THREAD

**Issue by jessebond2** [fluent/fluent-logger-node#101]
I'm getting this error when I try to setup the logger. Current on version 2.8.0

```
ERROR in ./node_modules/fluent-logger/lib/winston.js
Module not found: Error: Can't resolve 'winston-transport' in '/path/to/code/webpack/doc_page/node_modules/fluent-logger/lib'
 @ ./node_modules/fluent-logger/lib/winston.js 8:18-46
 @ ./node_modules/fluent-logger/lib/index.js
 @ ./src/shared/utils/fluent_monkey_patch.js
 @ ./src/index.server.jsx
```

**Comment by okkez** [comment]
Do you use fluent-logger's winston support?
It is optional, so we don't set the dependency for winston.

Maybe, you need to add the dependency for `winston@3.0.0` in your package.json.

**Comment by jessebond2** [comment]
Nope! Never even heard of that package before. Downgrading to 2.7.0 removed this issue.

**Comment by okkez** [comment]
Can you show me the reproducible code and package.json?

**Comment by fogine** [comment]
Confirming that the version `2.8.0` is broken - aka. not backward compatible.  
https://github.com/winstonjs/winston-syslog/issues/95

> winston-transport is not a dependency of Winston v2.x, only 3.0.

thus, when you are requiring `winston-transport` [HERE](https://github.com/fluent/fluent-logger-node/blob/master/lib/winston.js#L8) you effectively break all code which runs with winston  `v2.x) if i'm not mistaken.

To fix this issue, I suggest reverting the `2.8` release by releasing `2.8.1`.  You can then release your code with `winston 3.x` support only as `3.0.0`...

**Comment by okkez** [comment]
Users must set dependencies to `winston@2` or `winston@3` in their own package.json because fluent-logger's winston support is optional.

**Comment by fogine** [comment]
@okkez , that's not the issue.
Users have already installed `winston@2` / `winston@3` explicitly in their project. The problem is that `fluent-logger` prior version `2.8.0` worked with `winston@2`, however since version `2.8.0` of `fluent-logger` which [explicitly requires `winston-transport`](https://github.com/fluent/fluent-logger-node/blob/master/lib/winston.js#L8), it breaks the compatibility with `winston@2`  
From [winston-transport](https://github.com/winstonjs/winston-transport) project page:  

> Base stream implementations for winston@3 and up.

So based on the above quote, `fluent-logger` should not require its users to install the `winston-transport` package along basic `winston` package

**Comment by fogine** [comment]
This is an example of public project which fails to build since the `fluent-logger@2.8.0` release:  https://travis-ci.org/BohemiaInteractive/bi-logger/builds/395486808

**Comment by fogine** [comment]
@okkez , Okay, to fix projects which have in their package.json for example:  

```json
"dependencies": {
    "fluent-logger": "^2.4.0"
}
```

could we `npm unpublish fluent-logger@2.8.0` ?
Otherwise, npm will automatically update to `fluent-logger@2.8.0` which is not backward compatible...

**Comment by okkez** [comment]
@fogine If you want `fluent-logger@2.8.1` you can create the PR to 2.8 branch.
Could you create the PR for `fluent-logger@2.8.1`?
I will merge it and publish `fluent-logger@2.8.1`.

**Comment by okkez** [comment]
Released v2.8.1
